### PR TITLE
Use math native library

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

An evolution of native code management in math module.

**What is the current behavior?** *(You can also link to an open issue here)*

Native library has to be compiled separately and put in the path so that sparse matrix can be used.

**What is the new behavior (if this is a feature change)?**

powsybl-math-native jar has been uploaded to Maven central (see powsybl/powsybl-math-native repo). It contains native libraries for different platform and based on native libray loader project we can easily load native code from Maven central without any configuration.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: math/pom.xml [hunk 1]
```diff
@@ -48,6 +48,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.scijava</groupId>
+            <artifactId>native-lib-loader</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-math-native</artifactId>
+            <version>${powsyblmathnative.version}</version>
```
Comment (mathbagu) [inline math/pom.xml]: Should we declare the version in the parent pom.xml to be consistent with the rest of the project?

Comment (mathbagu) [reply math/pom.xml]: You also forgot to fix distribution-core/pom.xml to remove the copy of old libmathjni.so

Comment (geofjamg) [reply math/pom.xml]: > Should we declare the version in the parent pom.xml to be consistent with the rest of the project?
Indeed, it is an external dependency. Done.

Comment (geofjamg) [reply math/pom.xml]: > You also forgot to fix distribution-core/pom.xml to remove the copy of old libmathjni.so

Done

**Pull request merged**